### PR TITLE
test: fix 'incompatible pointer types' warnings

### PR DESCRIPTION
**Pull request by cjihrig**
Refs: https://github.com/libuv/libuv/pull/2686

### CODE REVIEW

File: test/test-poll-multiple-handles.c [hunk 1]
```diff
@@ -83,10 +83,10 @@ TEST_IMPL(poll_multiple_handles) {
   /* Closing an already stopped polling handle is safe in any case */
   uv_close((uv_handle_t*) &first_poll_handle, close_cb);
 
-  uv_unref(&second_poll_handle);
+  uv_unref((uv_handle_t*)&second_poll_handle);
```
Comment (vtjnash) [inline test/test-poll-multiple-handles.c:86]:
```suggestion
  uv_unref((uv_handle_t*) &second_poll_handle);
```

File: test/test-poll-multiple-handles.c [hunk 2]
```diff
@@ -83,10 +83,10 @@ TEST_IMPL(poll_multiple_handles) {
...
   ASSERT(0 == uv_run(uv_default_loop(), UV_RUN_DEFAULT));
   ASSERT(close_cb_called == 1);
-  uv_ref(&second_poll_handle);
+  uv_ref((uv_handle_t*)&second_poll_handle);
```
Comment (vtjnash) [inline test/test-poll-multiple-handles.c:89]:
```suggestion
  uv_ref((uv_handle_t*) &second_poll_handle);
```

**Review by vtjnash — APPROVED** [review]
LGTM (with style nit)

**Comment by cjihrig** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/2110/. None of the failures are related to the test changed in this PR.

**Review by santigimeno — APPROVED** [review]

**Comment by cjihrig** [comment]
Landed in https://github.com/libuv/libuv/commit/9c3d692b3941a2a4171629fb52af2e1029c415e8. Thanks for the reviews.

**Pull request closed without merging**